File: 3249-minimum-number-of-operations-to-make-array-xor-equal-to-k/minimum-number-of-operations-to-make-array-xor-equal-to-k.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        curr = 0
        for num in nums:
            curr ^= num
        count = 0
        while k or curr:
            if (curr%2) != (k%2):
                count += 1
            k//=2
            curr//=2
        return count

File: 3249-minimum-number-of-operations-to-make-array-xor-equal-to-k/minimum-number-of-operations-to-make-array-xor-equal-to-k.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        int curr_xor = 0;
        for(auto num: nums)
        {
            curr_xor ^= num;
        }
        int count = 0;
        while (k || curr_xor)
        {
            if((k%2) != (curr_xor % 2))
            {
                count +=1;
            }
            k/=2;
            curr_xor /= 2;
        }
        return count;
    }
};